packages/app/src/cli/services/dev.ts

```diff
@@ -163,6 +173,24 @@ interface DevFrontendTargetOptions extends DevWebOptions {
   backendPort: number
 }
 
+function devThemeExtensionTarget(
+  options: DevOptions,
+  apiKey: string,
+  store: string,
+  token: string,
+): ReverseHTTPProxyTarget {
+  return {
+    logPrefix: 'extensions',
+    action: async (_stdout: Writable, _stderr: Writable, _signal: abort.Signal, _port: number) => {
+      const adminSession = await session.ensureAuthenticatedAdmin(store)
+      const storefrontToken = await session.ensureAuthenticatedStorefront()
```

Comment:
Not sure about doing this here 🤔 
I think these tokens should be obtained at the start of `dev` (same as the partners token) if needed.

Admin tokens are a bit tricky and prone to errors because of the identity issue with development stores, better to throw that earlier than here where everything else is already running.

Reply:
That makes sense, it's better fail earlier indeed! Fixed :)